File: front end/src/app/officer/model/Person.ts
export class Person {
    private id: number;
    private name: string;
    private address: string;
    private nic: string;
    private dob: string;
    private gender: 'Male' | 'FEMALE';
    private maritalStatus: 'MARRIED' | 'SINGLE';
    private email: string;
    private phoneNumber: string;
    private occupation: string;
    private salary: number;
    private diseased: 'DIABETES' | 'HEART_DISEASE' | 'CANCER' | 'PARALYSIS' | 'ASTHMA' | 'NONE';
    private familyRole: 'MOTHER' | 'FATHER' | 'SON' | 'DAUGHTER';
    private familyId:number;
    private isActive: boolean;

    constructor(
        id: number,
        name: string,
        address: string,
        nic: string,
        dob: string,
        gender: 'Male' | 'FEMALE',
        maritalStatus: 'MARRIED' | 'SINGLE',
        email: string,
        phoneNumber: string,
        occupation: string,
        salary: number,
        diseased: 'DIABETES' | 'HEART_DISEASE' | 'CANCER' | 'PARALYSIS' | 'ASTHMA' | 'NONE',
        familyRole: 'MOTHER' | 'FATHER' | 'SON' | 'DAUGHTER',
        familyId:number,
        isActive: boolean
    ) {
        this.id = id;
        this.name = name;
        this.address = address;
        this.nic = nic;
        this.dob = dob;
        this.gender = gender;
        this.maritalStatus = maritalStatus;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.occupation = occupation;
        this.salary = salary;
        this.diseased = diseased;
        this.familyRole = familyRole;
        this.familyId=familyId;
        this.isActive = isActive;
    }
} 

File: back end/src/main/java/edu/e_g/dto/Person.java
package edu.e_g.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import edu.e_g.util.DiseasedType;
import edu.e_g.util.GenderType;
import edu.e_g.util.IsActiveType;
import edu.e_g.util.maritalStatusType;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Person {
    @JsonProperty("personId")
    private Long personId;

    @NotNull
    private String name;

    @NotNull
    private String address;

    @Pattern(regexp = "^\\d{9,10}[Vv]?$" ,message = "Please give 10 Digits or 'V' ")
    private String nic;

    @NotNull
    private LocalDate dob;

    @NotNull
    private GenderType gender;

    @JsonProperty("maritalStatus")
    @NotNull
    private maritalStatusType maritalStatus;

    @Email(message = "Please enter a valid email")
    private String email;

    @JsonProperty("phoneNumber")
    @Pattern(regexp = "^(\\+|0)\\d{9,12}$",message ="Please enter [ (10 Digits) or (+ and country number) ]")
    private String phoneNumber;

    @NotNull
    private String occupation;

    @NotNull
    private Double salary;

    @NotNull
    private DiseasedType diseased;

    private Long family;

    @JsonProperty("isActive")
    private IsActiveType isActive;
}
